### PR TITLE
PLAT-54708: Fixed `storybook` lists to show items properly when `dataSize` is greater than 1000.

### PR DESCRIPTION
**Pull request by 0x64**
### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
In `VirtualList` and `VirtualGridList` samples (for both `ui` and `moonstone`), if `dataSize` is set to any number grater than 1000, no item is rendered after 'Item 999'. 

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Data were initially generated with the size of 1000 and was never updated even if `dataSize` was changed.
To support any number of items, items will be re-generated whenever `dataSize` is changed. It is a natural behavior since the change of  `dataSize` means the change of data in real apps.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
Heading '0's for list subjects will be changed along with the number of digits to show. (For example, there will be no '0' if a list shows items from '0' to '9')

### Links
[//]: # (Related issues, references)
PLAT-54708

### Comments
Enact-DCO-1.0-Signed-off-by: Seungcheon Baek (sc.baek@lge.com)

### CODE REVIEW

File: packages/sampler/stories/moonstone-stories/VirtualGridList.js [hunk 1]
```diff
@@ -51,16 +52,28 @@ const
...
 
-for (let i = 0; i < 1000; i++) {
+const updateDataSize = (dataSize) => {
 	const
-		count = ('00' + i).slice(-3),
-		text = `Item ${count}`,
-		subText = `SubItem ${count}`,
-		color = Math.floor((Math.random() * (0x1000000 - 0x101010)) + 0x101010).toString(16),
-		source = `http://placehold.it/300x300/${color}/ffffff&text=Image ${i}`;
+		itemNumberDigits = dataSize > 1 ? Math.floor(Math.log10(dataSize - 1)) + 1 : 0,
```
Comment (ybsung) [inline packages/sampler/stories/moonstone-stories/VirtualGridList.js]: How about (dataSize - 1 + '').length? In addition, we need to check if dataSize equals 0 or not.

File: packages/sampler/stories/moonstone-stories/VirtualGridList.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ const
 		direction: {'horizontal': 'horizontal', 'vertical': 'vertical'}
 	},
 	items = [],
+	dataSizeDefault = 1000,
```
Comment (ybsung) [inline packages/sampler/stories/moonstone-stories/VirtualGridList.js]: How about `defaultDataSize`?

File: packages/sampler/stories/qa-stories/VirtualList.js [hunk 3]
```diff
@@ -34,9 +35,21 @@ const
...
-}
+const updateDataSize = (dataSize) => {
+	const
+		itemNumberDigits = dataSize > 0 ? ((dataSize - 1) + '').length : 0,
+		headingZeros = Array(itemNumberDigits).join('0');
+
+	items.length = 0;
+
+	for (let i = 0; i < dataSize; i++) {
+		items.push({item :'Item ' + (headingZeros + i).slice(-itemNumberDigits), selected: false});
```
Comment (ybsung) [inline packages/sampler/stories/qa-stories/VirtualList.js]: It seems that we need to initialize the `StatefullSwitchItem`'s `selected` state when updating `items` array.

Comment (ybsung) [reply packages/sampler/stories/qa-stories/VirtualList.js]: `items[this.props.index].selected` and `this.state.selected` are not matched each other in `StatefulSwitchItem `.

File: packages/sampler/stories/moonstone-stories/VirtualGridList.js [hunk 4]
```diff
@@ -51,16 +52,28 @@ const
...
+			count = (headingZeros + i).slice(-itemNumberDigits),
+			text = `Item ${count}`,
+			subText = `SubItem ${count}`,
+			color = Math.floor((Math.random() * (0x1000000 - 0x101010)) + 0x101010).toString(16),
+			source = `http://placehold.it/300x300/${color}/ffffff&text=Image ${i}`;
+
+		items.push({text, subText, source});
+	}
+
+	return dataSize;
```
Comment (ybsung) [inline packages/sampler/stories/moonstone-stories/VirtualGridList.js]: I could see the following warnnings.
```
The specified value "NaN" is not a valid number. The value must match to the following regular expression: -?(\d+|\d+\.\d+|\.\d+)([eE][-+]?\d+)?
```
Do we need to return 0 if `dataSize` equals null?

Comment (0x64) [reply packages/sampler/stories/moonstone-stories/VirtualGridList.js]: I've tested all changed samples with various values including 'no value', but I could not see that error. Could you let me know how can I reproduce it?

File: packages/sampler/stories/qa-stories/VirtualList.js (hunk 3 above)
Comment (0x64) [reply packages/sampler/stories/qa-stories/VirtualList.js]: Oh, you're right. Hmm. any better way to clear up states of all items?

Comment (0x64) [reply packages/sampler/stories/qa-stories/VirtualList.js]: Note: discussed with @MikyungKim and @ybsung, and concluded not to update qa-sampler. It is unclear to update all switch items when `dataSize` is changed, and qa-sampler is to test some special cases, not for general samples. I will revert changes of qa-sampler.

File: packages/sampler/stories/moonstone-stories/VirtualGridList.js (hunk 4 above)
Comment (0x64) [reply packages/sampler/stories/moonstone-stories/VirtualGridList.js]: It turns out that it is not from our code but from a knob-addon.

**Comment by ybsung** [comment]
:LGTM

@0x64 Please fix the conflict.

**Comment by MikyungKim** [comment]
LGTM

**Pull request merged**